modules/compute/vm-instance/variables.tf

```diff
@@ -245,6 +245,12 @@ variable "guest_accelerator" {
   default = null
 }
 
+variable "automatic_restart" {
+  description = "Specifies if the instance should be restarted if it was terminated by Compute Engine (not a user). Defaults to true."
+  type        = bool
+  default     = null
```

Comment:
This raises the same concerns as in #949. "Bool should not default to null; Bool should not be compared to null". I need to think on it some more as this case is somewhat simpler, but I wanted to raise this early so you could think about it as well.

Reply:
`add_deployment_name_before_prefix`  case is indeed simpler, because we have full understanding of its behaviour and doesn't leave the scope of the module.

`automatic_restart`  is different story, I would let to provider to fall to it's default behaviour while allowing user to be explicit if they want to.

We need `optional[bool]`, not `bool`, and `default = null` achieves it.